analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/Extensions/IOperationExtensions.cs

```diff
@@ -22,17 +22,22 @@ namespace SonarAnalyzer.SymbolicExecution.Roslyn
 {
     internal static class IOperationExtensions
     {
-        internal static ISymbol TrackedSymbol(this IOperation operation) =>
-            operation?.Kind switch
+        internal static ISymbol TrackedSymbol(this IOperation operation)
+        {
+            return operation?.Kind switch
             {
-                OperationKindEx.Conversion => TrackedSymbol(operation.ToConversion().Operand),
-                OperationKindEx.FieldReference when IFieldReferenceOperationWrapper.FromOperation(operation) is var fieldReference && IsStaticOrThis(fieldReference) => fieldReference.Field,
-                OperationKindEx.LocalReference => ILocalReferenceOperationWrapper.FromOperation(operation).Local,
-                OperationKindEx.ParameterReference => IParameterReferenceOperationWrapper.FromOperation(operation).Parameter,
-                OperationKindEx.Argument => IArgumentOperationWrapper.FromOperation(operation).Value.TrackedSymbol(),
+                OperationKindEx.Conversion when !IsTryDownCast(operation.ToConversion()) => TrackedSymbol(operation.ToConversion().Operand),
+                OperationKindEx.FieldReference when operation.ToFieldReference() is var fieldReference && IsStaticOrThis(fieldReference) => fieldReference.Field,
```

Comment:
This should be more consistent. Either capture the wrapper in both cases or none.
Generally speaking, it would be nice to have a `ToWrapper` method that can be used like so:
```
return operation.ToWrapper() switch
{
    IConversionOperationWrapper conversion when !IsTryDownCast(conversion) => TrackedSymbol(conversion.Operand),
    IFieldReferenceOperationWrapper fieldReference when IsStaticOrThis(fieldReference) => fieldReference.Field,
    ILocalReferenceOperationWrapper localReference => localReference.Local,
    IParameterReferenceOperationWrapper parameterReference => parameterReference.Parameter,
    IArgumentOperationWrapper argument => argument.Value.TrackedSymbol(),
    _ => null
};
```
```
internal static IOperationWrapper ToWrapper(this IOperation operation) =>
    operation?.Kind switch
    {
        OperationKindEx.Conversion => operation.ToConversion(),
        OperationKindEx.FieldReference => operation.ToFieldReference(),
        OperationKindEx.LocalReference => operation.ToLocalReference(),
        OperationKindEx.ParameterReference => operation.ToParameterReference(),
        OperationKindEx.Argument => operation.ToArgument(),
        // TODO add more cases
        _ => null
    };
```

Reply:
I've captured the `conversion`, that was a good catch of unintended code.

The switch on the wrappers doesn't work, unfortunately. That would make life way easier.